### PR TITLE
#2702 allowing validation query to be null

### PR DESCRIPTION
**Pull request by slivkamiro**
###### Problem:
tomcat jdbc allows validationQuery to be null, see https://tomcat.apache.org/tomcat-9.0-doc/jdbc-pool.html . In that case it will validate connections by isValid() method on Connection object.

###### Solution:
making validation query Optional. healthchecks calling isValid when validation query is not defined.

### CODE REVIEW

File: dropwizard-db/src/main/java/io/dropwizard/db/DataSourceFactory.java [hunk 1]
```diff
@@ -306,6 +306,9 @@
  * </table>
  */
 public class DataSourceFactory implements PooledDataSourceFactory {
+
+    public static final String DEFAULT_VALIDATION_QUERY = "/* Health Check */ SELECT 1";
```
Comment (jplock) [inline dropwizard-db/src/main/java/io/dropwizard/db/DataSourceFactory.java]: can this be `private`?

Comment (slivkamiro) [reply dropwizard-db/src/main/java/io/dropwizard/db/DataSourceFactory.java]: Done

**Comment by jplock** [comment]
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project dropwizard-hibernate: Compilation failure
[ERROR] /home/circleci/dropwizard/dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/HibernateBundle.java:[82,79] error: incompatible types: Optional<String> cannot be converted to String
```

**Comment by slivkamiro** [comment]
`Received 'killed' signal` not sure what happened there.

**Comment by jplock** [comment]
@joschi @nickbabcock thoughts on adding this to 2.0.0?

**Pull request merged**